Element84/filmdrop-ui
# Ba/feature/add cart buttons to count

**Related Issue(s):**

- NA

**Proposed Changes:**

1. Add buttons for: loading all scenes, selecting scenes per mock if more than initial load of 200 scenes 
> NOTE: max scenes added upper limit set to be 1k scenes based on limitations in API & rendering engine - discussed and approved with product owner
2. Add get all scenes service
3. Separate `imageOverlayLoading` state from `SearchLoading` state
4. Add `mappedScenes` to redux state to track all scenes on map not just a single search result
5. Add `clearExisting` param in `addDataToLayer()` function to keep DRY but allow for persisting scenes when adding more to map

Bonus 1: fix log error with update to datetimerange picker from `US` to `gregory`
Bonus 2: Migrate vitest coverage provider to use v8 instead of c8

### To test:

**confirm all automated tests pass** 🍏 `npm run test`

**Test buttons conditional render**
- load app
- click search
- zoom in until individual scenes show on map & senes returned are less than 200
- confirm `Select scenes` button renders in scene count at top center of map
- confirm `Load all scenes` button does not render in scene count at top center of map
- change search filters so that more than 200 scenes are returned

  > easy way to do this is set starting month back a few months
- confirm `Select scenes` and `Load all scenes` buttons render in scene count at top center of map
- zoom out to scene aggregate level for grid (not hex)
- confirm `Select scenes` and `Load all scenes` buttons do not render anymore
- change collection to something that uses hex aggressions (like Cop DEM)
- click search
- confirm `Select scenes` and `Load all scenes` buttons still do not render

**Test selecting scenes**
- load app
- zoom in until individual scenes show on map
- confirm `Select scenes` button shows in scene count at top center of map
- click `Select scenes` button
- confirm all scenes showing on map are selected 
- confirm `Select scenes` button is no longer selectable
- confirm popup result count reflects the number of selected scenes
- use arrow buttons in popup result and confirm connection between mapped scene highlight & overlay and popup result changes
- click 'x' close button in top right of popup result
- confirm popup closes
- confirm all scenes get deselected 

**Test loading all/more scenes**
- load app
- zoom in until individual scenes show on map
- change search filters so that more than 1000 scenes are returned
- click search
- click 'Load all scenes' button
- confirm button text changes to `cancel`
- confirm loading spinner shows next to button
- confirm loading spinner shows over top of map
- confirm footprints clear initially
- confirm footprints start getting re-added to map
- confirm this keeps happening until 1000 scenes get loaded
- confirm button text changes to say `Max scenes loaded`
- click search again
- before all scenes reload, click `cancel` button
- confirm button text changes back to `Load all scenes`
- confirm partially returned search results still persist on the map
- click `Load all scenes` button again
- confirm all scenes get cleared from map and loading all starts over again clean

**Test buttons don't render with autosearch**
- set `ADVANCED_SEARCH_ENABLED` to be `false` in config.json
- load app
- change search filters so that more than 200 scenes are returned
- click search
- zoom in until individual scenes show on map
- confirm `Select scenes` & `Load all scenes` buttons shows in scene count at top center of map
- toggle auto search on in search controls above map
- confirm `Select scenes` & `Load all scenes` buttons no longer render 


**PR Checklist:**
